src/devices/ant_antplus.c

```diff
@@ -0,0 +1,198 @@
+layer is not documented anywhere. ANT chips use an 8 byte key to generate a 2 byte
+network ID using an unspecified algorithm. Valid keys are only assigned by Garmin and
+require specific licensing terms (and in some cases a payment)
+ANT+ uses the basic ANT message structure but is a managed network with a specific
+network key and well defined device types, each sending "data pages" of 8 bytes
+with specific data for the device type. Most ANT+ devices are sports focused like
+heart rate monitors, bicycle sensors, or environmental sensors.
+
+Please note that unlike most devices in the rtl_433 repository, ANT+ devices
+operate in the ISM band between 2.4GHz and 2.5GHz. Decoding these signals
+requires an SDR capable of operating above 2.4GHz (e.g. PlutoSDR) or the
+use of a downconverter for rtl_sdr. Finally, since the protocol encodes at
+1Mbps, sampling rate should be -s 4M or higher. It can work with a sampling
+rate as low as 2Msps, but unreliably.
+
+Without knowing the 8 byte key, existing ANT chips do not allow sniffing
+of the data packets. Hence the need for this decoder. At the moment it's
+not possible to recover the ANT key from the 16 bit on-air network key.
+
+This decoder only captures and displays the low level packets, identifying the
+2 byte network key plus all other device characteristics and 8 byte ANT payload.
+It identifies ANT+ packets using the unique network key used (0xa6c5)
+Refer to ANT+ documentation for ech specific device to parse the content of
+the pages sent as 8 byte ANT+ payload.
+
+The ANT protocol is using an uncommon strategy for the premable: either 0x55 or 0xaa,
+depending on the value of the first bit of the following byte (0 and 1 respectively)
+nRF24L01, on which the ANT protocol is based, uses the same premable strategy.
+In order to determine if the packet is using 0x55 or 0xaa, both packets are extracted
+and the only valid one is determined by matching the packet CRC with a calculated value
+
+The unpacked payload is 18 bytes long structured as follows:
+
+    PNNDDTXLPPPPPPPPCC
+
+- P: Preamble: either 0x55 or 0xAA, depending on the value of first bit of the next byte
+- N: Network key, assume LSB first (ANT+ uses 0xc5a6, most invalid keys 0x255b)
+- D: Device number, 16 bit. LSB first
+- X: Transmission type
+- L: ANT payload length including CRC
+- P: 8 byte ANT or ANT+ payload
+- C: 16 bit CRC (CRC-16/CCITT-FALSE)
+
+L is always 10, because at the moment the ANT payload is always 8 bytes
+
+CREDITS:
+https://github.com/sghctoma/antfs-poc-defcon24
+https://reveng.sourceforge.io/ to reverse engineer the CRC algorithm used
+*/
+
+#include "decoder.h"
+
+static uint16_t crc16ccitt(uint8_t *data, size_t length)
+{
+    int i;
+    uint16_t crc = 0xffff;
+
+    while (length--) {
+        crc ^= *(unsigned char *)data++ << 8;
+        for (i = 0; i < 8; ++i)
+            crc = crc & 0x8000 ? (crc << 1) ^ 0x1021 : crc << 1;
+    }
+    return crc & 0xffff;
+}
```

Comment:
We should have proper native crc functions matching this.

Reply:
I must have missed it, sorry, not familiar with the code yet.

Reply:
Sometimes I get sad when I see the amount of code that represent a decoder. You spent so much time and its only a few lines of code. But then again it is very easy to maintain.